sandbox/libs/authn/src/main/java/org/opensearch/authn/realm/InternalRealm.java

```diff
@@ -117,6 +137,20 @@ protected AuthenticationInfo doGetAuthenticationInfo(AuthenticationToken token)
                 // Bad password
                 throw new IncorrectCredentialsException(INCORRECT_CREDENTIALS_MESSAGE);
             }
+        } else if (token instanceof BearerToken) {
+            JwtToken jwtToken;
+
+            // Verify the validity of JWT token
+            try {
+                jwtToken = JwtVerifier.getVerifiedJwtToken(((BearerToken) token).getToken());
+            } catch (RuntimeException e) {
```

Comment:
This would also swallow an NPE and label it as incorrect credentials.

Reply:
I think you mean because it is not BadCredentialsException--check above and let me know what you think.